internal/internal_utils.go

```diff
@@ -196,8 +197,12 @@ func newChannelContextHelper(
 }
 
 // GetWorkerIdentity gets a default identity for the worker.
+//
+// This contains a random UUID, generated each time it is called, to prevent identity collisions when workers share
+// other host/pid/etc information.  These alone are not guaranteed to be unique, especially when Docker is involved.
+// Take care to retrieve this only once per worker.
 func getWorkerIdentity(tasklistName string) string {
-	return fmt.Sprintf("%d@%s@%s", os.Getpid(), getHostName(), tasklistName)
+	return fmt.Sprintf("%d@%s@%s@%s", os.Getpid(), getHostName(), tasklistName, uuid.New())
```

Comment:
Only 3 callers exist currently, all of them are a single call when initializing a client or worker:

https://github.com/uber-go/cadence-client/blob/7c70757e7c7fab5e3f7fc8bc4a9c5a3acef01a8b/internal/client.go#L518-L521

https://github.com/uber-go/cadence-client/blob/7c70757e7c7fab5e3f7fc8bc4a9c5a3acef01a8b/internal/client.go#L565-L568

https://github.com/uber-go/cadence-client/blob/7c70757e7c7fab5e3f7fc8bc4a9c5a3acef01a8b/internal/internal_worker.go#L211-L213
(this one is called in several locations, e.g. `newActivityWorker`, `newAggregatedWorker`, `newSessionWorker`, `newWorkflowTaskWorkerInternal`)

Reply:
Starting it up locally does show both workers on the same UUID, in any case:
<img width="1203" alt="Screen Shot 2021-09-28 at 19 08 28" src="https://user-images.githubusercontent.com/77197/135191189-7072c539-9953-4c36-a1bc-1cfebb8f8ccc.png">

Reply:
Have you considered generating uuid once in the `init()` and then use it? It may be safer a safer way to prevent double generation.

Reply:
tbh I think double-generation may actually be desirable - what if they start the same worker twice in the same process? Either with a partial shutdown and starting it back up, or just running two concurrently.  E.g. the Kafka ingester we have works somewhat like this.

Reply:
Yeah, in that case it makes sense.